src/utils/common.ts

```diff
@@ -90,6 +93,18 @@ export const buildUrlOptions = (options: { [key: string]: string }): string => {
   return optionStr;
 };
 
+export const getOfferItemIndexByBestOffer = (gasUsd: (number | undefined)[], receiveAmount: number[]) => {
+  let index = 0;
+  let minAmount = gasUsd[0] ? receiveAmount[0] - gasUsd[0] : 100000;
+
+  for (let i = 1; i < gasUsd.length; i++) {
```

Comment:
replace by `gasUsd.forEach`?

Reply:
yeah but, I was saving one iteration here

Reply:
wdym one iteration? `forEach` returns index too

Reply:
also u can just go through offers array instead of passing prices/amounts and then by going through offers identify best price

to my view by doing this u can do it in single cycle instead of 3 (getting amounts, getting prices, getting best index)